addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -406,6 +406,8 @@ def _rtc_invite_members(self, member_ids=None):
             channel_member_domain = expression.AND([channel_member_domain, [('id', 'in', member_ids)]])
         members = self.env['discuss.channel.member'].search(channel_member_domain)
         for member in members:
+            if member.partner_id.im_status == "do_not_disturb":
+                member.sudo()._rtc_leave_call()
```

Comment:
Are they really in the call at this point? I'd say they are not since the domain above checks for members that *do not have any rtc session*. It feels off to search for them only to make them leave or ignore them later on. I think we should add another leaf to the search domain to avoid returning them.